### PR TITLE
ci: Switch Protobuf checks to non-deprecated bufbuild/buf-action

### PR DESCRIPTION
**Pull request by notpeter**
Switch to [bufbuild/buf-action](https://github.com/bufbuild/buf-action) as the existing actions we're using have been deprecated:
- `bufbuild/buf-breaking-action@v1`
- `bufbuild/buf-setup-action`

See: [Deprecation notice](https://github.com/bufbuild/buf-setup-action/commit/3de295e0428c275142bbc75a57488bb186e2a2f1) (2024-08-01).

Enabling `format` and `breaking_action` but disable `lint` as that is a bit aggressive.

Release Notes:

- N/A

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -65,11 +65,10 @@ jobs:
             echo "BUF_BASE_BRANCH=$GITHUB_BASE_REF" >> $GITHUB_ENV
           fi
 
-      - uses: bufbuild/buf-setup-action@v1
+      - name: Proto buf checks
```
Comment (maxdeviant) [inline .github/workflows/ci.yml:68]: "Protobuf" is one word:

```suggestion
      - name: Protobuf checks
```

Comment (notpeter) [reply .github/workflows/ci.yml]: Lol. I was going for the precision pun as the tool is called 'buf'. Obviously fell flat.

**Comment by notpeter** [comment]
I may break this into multiple PRs.  One whitespace only commit (`buf format -w`) for `.git-blame-ignore-revs` and then any actual changes in another PR.

File: crates/proto/proto/zed.proto [hunk 2]
```diff
@@ -4,2497 +4,2496 @@ package zed.messages;
 // Looking for a number? Search "// current max"
 
 message PeerId {
-    uint32 owner_id = 1;
-    uint32 id = 2;
+  uint32 owner_id = 1;
```
Comment (maxdeviant) [inline crates/proto/proto/zed.proto:7]: Is there a way we can keep the protobuf file formatted with 4 spaces? Feels like that matches up with the rest of our code, which all uses 4 spaces.

**Comment by notpeter** [comment]
Couldn't get the core functionality we want (detect breaking changes against branch merge target) so giving up for now.

**Pull request closed without merging**